OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1875,6 +1851,50 @@ bool Model::getAllControllersEnabled() const{
 void Model::setAllControllersEnabled( bool enabled ) {
     _allControllersEnabled = enabled;
 }
+bool Model::isControlsCacheValid(const SimTK::State& state) const{
+    return getDefaultSubsystem().isCacheValueRealized(state,_modelControlsIndex);
+}
+//------------------------------------------------------------------------------
+//            OVERRIDDEN METHOD TO INITIALIZE CONTROLS CACHE
+//------------------------------------------------------------------------------
+void Model::extendRealizeTopology(SimTK::State& s) const
+{    
+    Super::extendRealizeTopology(s);
+
+    Model *mutableThis = const_cast<Model *>(this);
+
+    // Create the shared cache that will hold all model controls
+    // This must be created before Actuator.extendAddToSystem() since Actuator will append 
+    // its "slots" and retain its index by accessing this cached Vector
+    // value depends on velocity and invalidates dynamics BUT should not trigger
+    // recomputation of the controls which are necessary for dynamics
+    mutableThis->_modelControlsIndex = getDefaultSubsystem().allocateCacheEntry(
+        s, Stage::Velocity, Stage::Acceleration, new Value<Vector>());
+}
```

Comment:
the dependsOn `Stage::Velocity` makes sense but `Stage::Acceleration` for the `computeBy` stage, seems strange.  My guess is that we do not need controls on muscles to be computed in order to evaluate muscle forces (in general). That is the current force of a muscle with activation dynamics and/or tendon compliance is completely determined by the tendon stretch or its activation state. In which case the controls are only necessary to evaluate the derivatives which occurs at `Stage::Acceleration`.